File: practicas/bibli/SistemaBiblioteca.java
package practicas.bibli;
import java.util.ArrayList;


public class SistemaBiblioteca {
    public static void main(String[] args) {
        // Crear horario
        horario horario = new horario("Lunes a Viernes", "08:00", "18:00");

        // Crear biblioteca
        biblioteca biblioteca = new biblioteca("Biblioteca UMSA", horario);

        // Crear autores
        autor autor1 = new autor("Gabriel Garcia Marquez", "Colombiano");
        autor autor2 = new autor("Isabel Allende", "Chilena");
        biblioteca.agregarAutor(autor1);
        biblioteca.agregarAutor(autor2);

        // Crear libros
        ArrayList<pagina> paginasLibro1 = new ArrayList<>();
        libro libro1 = new libro("Cien Años de Soledad", "ISBN001", paginasLibro1);
        libro1.agregarPagina(new pagina(1, "Capítulo 1..."));
        libro1.agregarPagina(new pagina(2, "Capítulo 2..."));
        ArrayList<pagina> paginasLibro2 = new ArrayList<>();
        libro libro2 = new libro("La Casa de los Espíritus", "ISBN002", paginasLibro2);
        libro2.agregarPagina(new pagina(1, "Capítulo 1..."));

        biblioteca.agregarLibro(libro1);
        biblioteca.agregarLibro(libro2);

        // Crear estudiante
        estudiante estudiante = new estudiante("S001", "Juan Perez");

        // Prestar libro
        biblioteca.prestarLibro(estudiante, libro1);

        // Mostrar estado
        biblioteca.mostrarEstado();

        // Cerrar biblioteca
        biblioteca.cerrarBiblioteca();
    }
}

File: practicas/biblio/SistemaBiblioteca.py
from biblioteca import Biblioteca
from libro import Libro
from autor import Autor
from estudiante import Estudiante
from pagina import Pagina
from horario import Horario

# Crear horario
horario = Horario("Lunes a Viernes", "08:00", "18:00")

# Crear biblioteca
biblioteca = Biblioteca("Biblioteca UMSA", horario)

# Crear autores
autor1 = Autor("Gabriel Garcia Marquez", "Colombiano")
autor2 = Autor("Isabel Allende", "Chilena")
biblioteca.agregar_autor(autor1)
biblioteca.agregar_autor(autor2)

# Crear libros
libro1 = Libro("Cien Años de Soledad", "ISBN001")
libro1.agregar_pagina(Pagina(1, "Capítulo 1..."))
libro1.agregar_pagina(Pagina(2, "Capítulo 2..."))

libro2 = Libro("La Casa de los Espíritus", "ISBN002")
libro2.agregar_pagina(Pagina(1, "Capítulo 1..."))

biblioteca.agregar_libro(libro1)
biblioteca.agregar_libro(libro2)

# Crear estudiante
estudiante = Estudiante("S001", "Juan Perez")

# Prestar libro
biblioteca.prestar_libro(estudiante, libro1)

# Mostrar estado
biblioteca.mostrar_estado()

# Cerrar biblioteca
biblioteca.cerrar_biblioteca()
